src/environment/global/celestial_information.cpp

```diff
@@ -156,7 +156,7 @@ std::string CelestialInformation::GetLogHeader() const {
     // Acquisition of body name from id
     bodc2n_c(planet_id, kMaxNameLength, name_buffer, (SpiceBoolean*)&found);
     std::string name = name_buffer;
-    std::transform(name.begin(), name.end(), name.begin(), ::tolower);
+    std::transform(name.begin(), name.end(), name.begin(), [](unsigned char c) { return (char)std::tolower(c); });
```

Comment:
`int std::tolower(int ch)` は C の関数であることによって型が微妙なので，`charT std::tolower(charT c, const locale& loc)` を使ってもよいかもしれません（そうでなくとも C ロケールによって挙動が変わるので明示的に指定しておいてよさそう）．

```suggestion
    std::locale loc = std::locale::classic();
    std::transform(name.begin(), name.end(), name.begin(), [](char c) { return std::tolower(c, loc); });
```

Reply:
ご提案ありがとうございます。ただ、Visual Studioでビルドエラー出てしまうようです。

https://github.com/ut-issl/s2e-core/actions/runs/4490262461/jobs/7897109428?pr=367

VSでは`_tolower_l`を使う必要があるみたいです。

https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/tolower-tolower-towlower-tolower-l-towlower-l?view=msvc-170

Reply:
@sksat VSだけ、`_tolower_l`を使うというのも微妙なので、`int std::tolower(int ch)`に戻すことも含めて議論したいですが、いかがでしょう？

Reply:
え，Visual Studio が標準に準拠してないやつでしたか......了解です

Reply:
いや，上のドキュメントは VC++ のではなく VC のものなので違う気がしますね．
`#include <locale>` を追加してもう一度試してみてもらえますか？

Reply:
`charT std::tolower(charT c, const locale& loc)` はこれのことです（C のライブラリを std 名前空間に押し込んだやつではなく，C++ のテンプレート関数として C++ の標準規格に入っているもの）．
https://cpprefjp.github.io/reference/locale/tolower.html

Reply:
もしそれでもダメな場合は仕方ないので `int std::tolower(int ch)` に戻すことも検討します．ただ，その場合でも C-style cast ではなく `static_cast` を使うようにしたいですかね．

Reply:
ありがとうございます。includeしていなかったことが原因だったようで、無事ビルドできました。(自分の手元にない環境のデバッグは難しいですね。。。)

Reply:
お，よかったです．
そうなんですよね...... > 自分の手元に無い環境のデバッグ